File: scripts/check_db.py
import psycopg2
import os

try:
    conn = psycopg2.connect(
        host='aws-1-ap-southeast-1.pooler.supabase.com',
        port='5432',
        dbname='postgres',
        user='postgres.likjiuylgndoedpkzxhb',
        password=os.environ.get('SUPABASE_DB_PASSWORD', ''),
        sslmode='require'
    )
    cur = conn.cursor()
    # Check if table exists
    cur.execute("SELECT EXISTS (SELECT FROM pg_tables WHERE schemaname = 'public' AND tablename = 'comparisons');")
    exists = cur.fetchone()[0]
    print(f'Table exists: {exists}')
    
    if exists:
        cur.execute("SELECT COUNT(*) FROM comparisons")
        count = cur.fetchone()[0]
        print(f'Total comparisons in DB: {count}')
        
        cur.execute("SELECT request_id, endpoint, created_at FROM comparisons ORDER BY created_at DESC LIMIT 5")
        rows = cur.fetchall()
        print('Latest 5 comparisons:')
        for row in rows:
            print(row)
            
        cur.execute("INSERT INTO comparisons (request_id, endpoint, method, severity) VALUES ('req-py-test-999', '/api/test', 'GET', 'low') ON CONFLICT (request_id) DO NOTHING")
        conn.commit()
        print('Inserted a test row directly via psycopg2')
        
        cur.execute("SELECT COUNT(*) FROM comparisons")
        count2 = cur.fetchone()[0]
        print(f'Total comparisons in DB after insert: {count2}')
    
    conn.close()
except Exception as e:
    print('Error:', e)

File: scripts/check_db.js
const { Client } = require('pg');
const client = new Client({
  connectionString: 'postgres://postgres.likjiuylgndoedpkzxhb:' + process.env.SUPABASE_DB_PASSWORD + '@aws-0-ap-southeast-1.pooler.supabase.com:6543/postgres'
});
client.connect().then(async () => {
  try {
    const res = await client.query('SELECT COUNT(*) FROM comparisons');
    console.log('Total rows superuser sees:', res.rows[0].count);
    const rows = await client.query('SELECT request_id FROM comparisons LIMIT 2');
    console.log('Sample requests:', rows.rows);
    const policies = await client.query(`SELECT polname, cmd FROM pg_policy WHERE tablename = 'comparisons'`);
    console.log('Policies on table:', policies.rows);
  } catch (e) {
    console.error('Error querying DB:', e.message);
  }
  client.end();
}).catch(e => {
  console.error('Connection error:', e.message);
  process.exit(1);
});
